resolvewithplus.js

```diff
@@ -183,33 +183,67 @@ const getesmkeyvalglobreplaced = (esmkey, esmval, pathlocal) => {
 //
 //   All instances of * on the right hand side will then be replaced
 //   with this value, including if it contains any / separators.
-const getesmkeyvalmatch = (esmkey, esmval, path, keyvalmatch = false) => {
-  if (ispathesmmatch(esmkey, path)) {
-    if (esmval.includes('*')) {
-      if (ispathesmmatch(esmval, path)) {
-        keyvalmatch = path
+const getesmkeyvalmatch = (esmkey, esmval, idpath, opts, keyvalmx = false) => {
+  if (ispathesmmatch(esmkey, idpath)) {
+    if (String(esmval).includes('*')) {
+      if (ispathesmmatch(esmval, idpath)) {
+        keyvalmx = idpath
       } else if (esmkey.includes('*') && esmkey !== esmval) {
-        keyvalmatch = getesmkeyvalglobreplaced(esmkey, esmval, path)
+        keyvalmx = getesmkeyvalglobreplaced(esmkey, esmval, idpath)
       }
     } else {
-      keyvalmatch = esmval
+      // if below condition is true, assume exports look this way and,
+      //   * the namespace defined on the key is valid for the moduleId,
+      //   * expanded key used as replacement for nested path value wildcard
+      // ```
+      // "exports": {
+      //   "./*": {
+      //     "default": "./src/*/index.js",
+      //     "types": "./types/*/index.d.ts"
+      //   }
+      // }
+      // ```
+      if (isobj(esmval) && esmkey.includes('*')) {
+        const resolvedkey = idpath // 'mystuff'
+        const expandedkey = path.posix.join(idpath, esmkey) // 'mystuff/*'
+        const expandedspec = Object.keys(esmval).reduce((exp, nestkey) => {
+          const pathfirstdir = esmval[nestkey] // ./src/a/b.js -> 'src'
+            .split(/\.?\//).find(e => e)
+
+          // eg,
+          // getesmkeyvalglobreplaced(
+          //   'mystuff/*', 'src/mystuff/*', 'mystuff/index.js')
+          // 'src/mystuff/index.js'
+          exp[nestkey] = getesmkeyvalglobreplaced(
+            expandedkey,
+            path.posix.join(pathfirstdir, expandedkey),
+            path.posix.join(resolvedkey, esmval[nestkey].split('*')[1]))
```

Comment:
I'm not sure this will support multiple occurrences of `*` on the "right side".

> All instances of `*` on the right hand side will then be replaced with this value, including if it contains any `/` separators.

```
// import { stuff } from 'my-module/stuff' <-- should resolve to "my-module/src/stuff/some-folder/stuff/index.js"
{
  "name": "my-module"
 "exports": {
    "./*": {
      "default": "./src/*/some-folder/*/index.js"
    }
}
```